packages/rocketchat-livechat/server/publications/livechatRooms.js

```diff
@@ -29,13 +29,35 @@ Meteor.publish('livechat:rooms', function(filter = {}, offset = 0, limit = 20) {
 			query.open = { $exists: false };
 		}
 	}
-	if (filter.from && filter.to) {
-		var StartDate = new Date(filter.from);
-		var ToDate = new Date(filter.to);
-		ToDate.setDate(ToDate.getDate() + 1);
-		query['ts'] = { $gt: StartDate, $lt: ToDate };
+	if (filter.from) {
+		query['ts'] = { $gte: filter.from };
 	}
+	if (filter.to) {
+		if (!query['ts']) {
+			query['ts'] = {};
+		}
+		filter.to.setDate(filter.to.getDate() + 1);
+		filter.to.setSeconds(filter.to.getSeconds() - 1);
+		query.ts.$lte = filter.to;
```

Comment:
I think this would be more elegant: `query.ts = _.extend(query.ts || {}, { $lte: filter.to })`, and you'd remove lines 36-38.